tmail-backend/tmail-third-party/open-ai/src/main/java/com/linagora/tmail/mailet/AIBotConfig.java

```diff
@@ -37,7 +35,7 @@ public class AIBotConfig {
     private final MailAddress gptAddress;
     private final LlmModel llmModel;
 
-    public AIBotConfig(String apiKey, @Nullable URL baseURL, MailAddress gptAddress, LlmModel llmModel) {
+    public AIBotConfig(String apiKey, MailAddress gptAddress, LlmModel llmModel, @Nullable URL baseURL) {
```

Comment:
I am unhappy that we take decisions in this constructor (default values). Maybe calling code can resolve the default value and just pass non nullable values here?

Reply:
That is exactly what is happening, `fromMailetConfig` calls the constructor with valid values, the assertions were added just in case.

Reply:
Then why do we have a @Nullable parameter?

Reply:
Made it Optional.